src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Text.Json;
+using AzureExtension.Contracts;
+using DevHomeAzureExtension.DataModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+
+namespace AzureExtension.DevBox;
+
+public class DevBoxProvider : IComputeSystemProvider, IDisposable
+{
+    private readonly IHost _host;
+
+    public DevBoxProvider(IHost host)
+    {
+        _host = host;
+    }
+
+    string IComputeSystemProvider.DefaultComputeSystemProperties
+    {
+        get => throw new NotImplementedException();
+        set => throw new NotImplementedException();
+    }
+
+    public string DisplayName => "DevBox Provider";
+
+    public string Id => "182AF84F-D5E1-469C-9742-536EFEA94630";
+
+    public string Properties => throw new NotImplementedException();
+
+    public ComputeSystemProviderOperation SupportedOperations => throw new NotImplementedException();
+
+    private bool IsValid(JsonElement jsonElement)
+    {
+        return jsonElement.ValueKind != JsonValueKind.Undefined;
+    }
+
+    public IEnumerable<IComputeSystem> GetComputeSystemsAsync(IDeveloperId? developerId)
+    {
+        var computeSystems = new List<IComputeSystem>();
+
+        var mgmtSvc = _host.Services.GetService<IDevBoxManagementService>();
+        if (mgmtSvc != null)
+        {
+            return computeSystems;
+        }
+        else
+        {
+            Log.Logger()?.ReportError($"Error getting systems: Rest Service not configured");
+            throw new ArgumentException($"Rest Service needs to be configured.");

```

Comment:
It seems annoying to me every time we do logging and throw exceptions, especially with the same text message. Can we do logging when we catch this?
Or may be use exception as ReportError argument to avoid text duplication:
var ex = new ArgumentException($"Error getting systems: Rest Service not configured");
Log.Logger()?.ReportError(ex);
throw ex;

Reply:
@dhoehna suggested not to throw exceptions at all, so I will be removing them in a future PR completely.